pinot-query-planner/src/main/java/org/apache/pinot/query/planner/physical/colocated/GreedyShuffleRewritePreComputeVisitor.java

```diff
@@ -64,4 +65,11 @@ public Integer visitTableScan(TableScanNode stageNode, GreedyShuffleRewriteConte
     context.addLeafNode(stageNode.getStageId(), stageNode);
     return 0;
   }
+
+  @Override
+  public Integer visitSetOp(SetOpNode setOpNode, GreedyShuffleRewriteContext context) {
```

Comment:
@ankitsultana can you take a look and see if this works with your expectation? even if we do not allow visitSetOp is a valid current limitation IMO

Reply:
I added a new field for SetOp. Partitioning is a must to ensure  the correctness.